src/UhdmAst.cpp

```diff
@@ -968,19 +968,10 @@ namespace UhdmAst {
           } else {
             AstNode* assign;
             if (lvalue->type() == AstType::en::atVar) {
-              // If a variable was declared along with the assignment,
-              // return it as well. Create a reference for the assignment.
-              AstNode* var = lvalue;
-              auto* var_ref = new AstParseRef(new FileLine("uhdm"),
-                                                     VParseRefExp::en::PX_TEXT,
-                                                     lvalue->name(),
-                                                     nullptr,
-                                                     nullptr);
-              if (objectType == vpiContAssign)
-                assign = new AstAssignW(new FileLine("uhdm"), var_ref, rvalue);
-              else
-                assign = new AstAssign(new FileLine("uhdm"), var_ref, rvalue);
-              var->addNextNull(assign);
+              // This is not a true assignment
+              // Set initial value to a variable and return it
+              AstVar* var = static_cast<AstVar*>(lvalue);
```

Comment:
I see that we already use regular casting throughout this file so I guess it is okay to keep it here too (at least it is consistent), but in general, Verilator uses the `VN_CAST` macro to check and cast and the `VN_IS` macro to just check the type of the node.
At some point we will probably have to convert that to use the macros if we wish to be integrated with mainline Verilator
(this is low prio of course)

Reply:
In addition, the general convention in verilator is to add `p` prefix if the variable is a pointer, so
`AstVar var;` would be ok, but for a pointer it should be `AstVar* varp;`
(This is also something that should be done in a bulk)

Reply:
Created https://github.com/antmicro/verilator/issues/35 for this task.